### PR TITLE
Fix drop doors after break

### PR DESCRIPTION
**Pull request by KorDum**
Fix drop doors after break
Improve code readability

### CODE REVIEW

**Comment by Elec332** [comment]
oops, wrong button

File: src/main/java/binnie/extratrees/block/BlockETDoor.java [hunk 1]
```diff
@@ -59,9 +63,9 @@ public IIcon getIcon(final int side, final int meta) {
 
 	@Override
 	@SideOnly(Side.CLIENT)
-	public IIcon getIcon(final IBlockAccess par1IBlockAccess, final int par2, final int par3, final int par4, final int par5) {
+	public IIcon getIcon(final IBlockAccess par1IBlockAccess, final int x, final int y, final int z, final int par5) {
```
Comment (marcin212) [inline src/main/java/binnie/extratrees/block/BlockETDoor.java]:
```java
final IBlockAccess par1IBlockAccess
```
Could you change this to: blockAccess

File: src/main/java/binnie/extratrees/block/BlockETDoor.java [hunk 2]
```diff
@@ -127,9 +131,9 @@ public int getRenderType() {
 
 	@Override
 	@SideOnly(Side.CLIENT)
-	public int colorMultiplier(final IBlockAccess par1IBlockAccess, final int par2, final int par3, final int par4) {
+	public int colorMultiplier(final IBlockAccess par1IBlockAccess, final int x, final int y, final int z) {
```
Comment (marcin212) [inline src/main/java/binnie/extratrees/block/BlockETDoor.java]: and here

File: src/main/java/binnie/extratrees/block/BlockETDoor.java [hunk 3]
```diff
@@ -166,41 +170,45 @@ else if (j1 == 3 && par5 == 2) {
...
 			return WoodManager.getPlankType(meta & 0xFF).getColour();
 		}
-		final int meta = TileEntityMetadata.getTileMetadata(par1IBlockAccess, par2, par3, par4);
+		final int meta = TileEntityMetadata.getTileMetadata(par1IBlockAccess, x, y, z);
 		return WoodManager.getPlankType(meta & 0xFF).getColour();
 	}
 
-	public int getFullMetadata(final IBlockAccess par1IBlockAccess, final int par2, final int par3, final int par4) {
-		final int l = par1IBlockAccess.getBlockMetadata(par2, par3, par4);
+	public int getFullMetadata(final IBlockAccess par1IBlockAccess, final int x, final int y, final int pzr4) {
```
Comment (marcin212) [inline src/main/java/binnie/extratrees/block/BlockETDoor.java]: same here

File: src/main/java/binnie/extratrees/block/BlockETDoor.java [hunk 4]
```diff
@@ -166,41 +170,45 @@ else if (j1 == 3 && par5 == 2) {
...
 	}
 
 	@Override
 	public ArrayList<ItemStack> getDrops(final World world, final int x, final int y, final int z, final int blockMeta, final int fortune) {
-		return BlockMetadata.getBlockDropped(this, world, x, y, z, blockMeta);
+		int yCoord = y;
+		if ((blockMeta & 0x08) != 0) {
+			yCoord -= 1;
+		}
+		return BlockMetadata.getBlockDropped(this, world, x, yCoord, z, blockMeta);
```
Comment (marcin212) [inline src/main/java/binnie/extratrees/block/BlockETDoor.java:191]: Did you test it?

**Review by marcin212 — CHANGES_REQUESTED** [review]

File: src/main/java/binnie/extratrees/block/BlockETDoor.java (hunk 4 above)
Comment (KorDum) [reply src/main/java/binnie/extratrees/block/BlockETDoor.java:191]: Yes. Door drops correctly if you cut it from up block or down block.

**Comment by KorDum** [comment]
I made some changes to the code.

File: src/main/java/binnie/extratrees/block/BlockETDoor.java (hunk 4 above)
Comment (KorDum) [reply src/main/java/binnie/extratrees/block/BlockETDoor.java:191]: I tested with different doors: from different materials and different metas.

**Review by marcin212 — APPROVED** [review]

**Comment by mezz** [comment]
Please check if this needs to be done for the 1.11.2 version as well

**Pull request merged**